apps/io-services-cms-webapp/src/utils/converters/service-lifecycle-converters.ts

```diff
@@ -35,28 +43,59 @@ export const payloadToItem = (
   },
 });
 
-export const itemToResponse = ({
-  fsm,
-  data,
-  id,
-  last_update,
-}: ServiceLifecycle.ItemType): ServiceResponsePayload => ({
-  id,
-  status: toServiceStatus(fsm),
-  last_update: last_update ?? new Date().getTime().toString(),
-  name: data.name,
-  description: data.description,
-  organization: data.organization,
-  metadata: {
-    ...data.metadata,
-    scope: toScopeType(data.metadata.scope),
-    category: toCategoryType(data.metadata.category),
-  },
-  require_secure_channel: data.require_secure_channel,
-  authorized_recipients: data.authorized_recipients,
-  authorized_cidrs: data.authorized_cidrs,
-  max_allowed_payment_amount: data.max_allowed_payment_amount,
-});
+export const itemToResponse =
+  (dbConfig: TopicPostgreSqlConfig) =>
+  ({
+    fsm,
+    data: {
+      metadata: { scope, category, topic_id, ...metadata },
+      ...data
+    },
+    id,
+    last_update,
+  }: ServiceLifecycle.ItemType): TE.TaskEither<
+    IResponseErrorInternal,
+    ServiceResponsePayload
+  > =>
+    pipe(
+      toServiceTopic(dbConfig)(id, topic_id),
+      TE.bimap(
+        (err) => ResponseErrorInternal(err.message),
+        (topic) => ({
+          id,
+          status: toServiceStatus(fsm),
+          last_update: last_update ?? new Date().getTime().toString(),
+          ...data,
+          metadata: {
+            ...metadata,
```

Comment:
topic_id is still returned?
In such case you can destructuring metadata 
`const { topic_id, ...otherMetadata } = metadata;`

and replacing ...metadata with ...otherMetadata

Reply:
no, it is not returned because metadata is already destructured in the following way:
`metadata: { scope, category, topic_id, ...metadata }`
If it is not clear enough, I can change the name of destructured `metadata` field to `otherMetadata`

Reply:
perfect!